lib/widgets/action_sidebar.dart

```diff
@@ -0,0 +1,219 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+class ActionSideBar extends StatefulWidget {
+  const ActionSideBar(this.actionButtons);
+
+  final List<Widget> actionButtons;
+
+  @override
+  _ActionSideBarState createState() => _ActionSideBarState();
+}
+
+class _ActionSideBarState extends State<ActionSideBar>
+    with SingleTickerProviderStateMixin {
+  _ActionSideBarState({this.isExtended = false});
+
+  bool isExtended;
+
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    _controller = AnimationController(
+      duration: const Duration(milliseconds: 250),
+      vsync: this,
+    );
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  void onTap() {
+    setState(() {
+      isExtended = !isExtended;
+
+      if (isExtended) {
+        _controller.forward();
+      } else {
+        _controller.reverse();
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (MediaQuery.of(context).size.width > 1100) {
```

Comment:
Maybe put this value in a const, you're using it in multiple places and risk forgetting to change all of them if you ever wanna change it.

Reply:
Extracted as a const value:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/8c0e90bd1db807c70ea5422527345d766711da4c#diff-91af2d9745e86244940959ec39c99cca9b8b43c1502042d85f16fcc6a311030bR20